panic error when create pipeline if no cloud is registered

**Environment**
- branch: v1-preview

**What happened**
When  create pipeline, there is panic error:
```
[2017-09-20 16:09:27] [INFO] [event.go:136] create service handler
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1fb46fc]

goroutine 12 [running]:
github.com/caicloud/cyclone/event.createServiceHandler(0xc420118300, 0xc4204824b0, 0xc4204ac2d0)
	/Users/robin/gocode/src/github.com/caicloud/cyclone/event/event.go:143 +0xec
github.com/caicloud/cyclone/event.handleEvent(0xc420118300, 0xc42030c000, 0x0)
	/Users/robin/gocode/src/github.com/caicloud/cyclone/event/event.go:98 +0x66
github.com/caicloud/cyclone/event.handlePendingEvents()
	/Users/robin/gocode/src/github.com/caicloud/cyclone/event/manager.go:406 +0xc3
created by github.com/caicloud/cyclone/event.Init
	/Users/robin/gocode/src/github.com/caicloud/cyclone/event/manager.go:77 +0x1da
```

Related code:
```go
	worker, err := CloudController.Provision(string(event.EventID), opts)
	if err != nil {
		return err
	}

	err = worker.Do()
	if err != nil {
		logdog.Error("run worker err", logdog.Fields{"err": err})
		return err
	}
```
When no cloud is registered, the worker is nil, which causes the panic error.

**What you expected to happen**
- If no cloud is registered, there will be error when provision worker.

--- a/cloud/controller.go
+++ b/cloud/controller.go
@@ -17,6 +17,7 @@ limitations under the License.
 package cloud
 
 import (
+	"errors"
 	"fmt"
 
 	"github.com/caicloud/cyclone/pkg/register"
@@ -114,6 +115,10 @@ func (cc *Controller) Provision(id string, opts WorkerOptions) (Worker, error) {
 
 	if cc.provisionErr == nil {
 		cc.provisionErr = NewErrCloudProvision()
+	}
+
+	if len(cc.Clouds) == 0 {
+		return nil, errors.New("Cloud needs to be registered first")
 	}
 
 	if opts.Quota.IsZero() {

